src/scripts/Viewer.vue

```diff
@@ -0,0 +1,110 @@
+<template>
+	<div class="wrapper">
+		<div class="viewer">
+			<button class="viewer--close" @click="close"></button>
+			<div class="viewer--container" :style="'background-image:url(' + imgSrc+ ')'"></div>
+		</div>
+	</div>
+</template>
+
+<script>
+const appName = require('../../package.json').name;
+
+export default {
+	name: "Viewer",
+	methods: {
+		close() {
+			this.$router.push('/');
+		}
+	},
+	computed: {
+		file() {
+			return _.findWhere(FileList.files, {
+				name: this.$route.params.file
+			});
+		},
+		dir() {
+			return window[appName].context.dir;
+		},
+		list() {
+			return _.filter(FileList.files, {
+				mimetype: "image/png"
+			});
+		},
+		thumbDimensions() {
+			let width = $(window).width();
+			switch (true) {
+				case (width <= 1024) :
+					return 1024
+					break;
+				case (width <= 1280) :
+					return 1280
+					break;
+				case (width <= 1920) :
+					return 1920
+					break;
+				default:
+					return 2160
+					break;
+			}
+		},
+		imgSrc() {
+			let path = [
+				OC.getRootPath(),
+				'remote.php/dav/files',
+				OC.getCurrentUser().uid + this.file.path,
```

Comment:
to be safe when handling paths, you can use `OC.joinPaths(a, b, c, ...)` which will correctly strip leading/trailing slashes to assemble the path